_upload_release/upload.go

```diff
@@ -0,0 +1,140 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	tcclient "github.com/taskcluster/taskcluster-client-go"
+	"github.com/taskcluster/taskcluster-client-go/auth"
+)
+
+const BUCKET_NAME = "downloads-taskcluster-net"
+const URL_PREFIX = "https://downloads.taskcluster.net"
+const OBJECT_PREFIX = "taskcluster-cli/"
+
+func getSTSCredentials() (id, secret, token string, err error) {
+	creds := tcclient.Credentials{}
+	client := auth.New(&creds)
+
+	// if running in a task, use the taskcluster proxy
+	if os.Getenv("TASK_ID") != "" {
+		client.BaseURL = "http://taskcluster/auth/v1"
+		log.Printf("Getting AWS credentials for %s/%s via taskcluster-proxy", BUCKET_NAME, OBJECT_PREFIX)
+	} else {
+		creds.ClientID = os.Getenv("TASKCLUSTER_CLIENT_ID")
+		creds.AccessToken = os.Getenv("TASKCLUSTER_ACCESS_TOKEN")
+		creds.Certificate = os.Getenv("TASKCLUSTER_CERTIFICATE")
+		log.Printf("Getting AWS credentials for %s/%s with clientId %s",
+			BUCKET_NAME, OBJECT_PREFIX, creds.ClientID)
+	}
+
+	resp, err := client.AwsS3Credentials("read-write", BUCKET_NAME, OBJECT_PREFIX, "")
+	if err != nil {
+		return "", "", "", err
+	}
+	log.Printf("Got AWS credentials, accessKeyId %s", resp.Credentials.AccessKeyID)
+	return resp.Credentials.AccessKeyID,
+		resp.Credentials.SecretAccessKey,
+		resp.Credentials.SessionToken,
+		nil
+}
+
+func getS3() (*s3.S3, error) {
+	id, secret, token, err := getSTSCredentials()
+	if err != nil {
+		return nil, err
+	}
+
+	creds := credentials.NewStaticCredentials(id, secret, token)
+	sess := session.Must(session.NewSession(aws.NewConfig().
+		WithRegion("us-west-1").
+		WithCredentials(creds)))
+
+	return s3.New(sess), nil
+}
+
+func exitErrorf(msg string, args ...interface{}) {
```

Comment:
oh, good point -- I'll make a followup PR (I had copy/pasted that from the AWS SDK examples)